std/typecons.d

```diff
@@ -1190,136 +1206,139 @@ if (distinctFieldNames!(Specs))
-                    }
-                }
-                else if (fmt.spec == 's')
+                if (fmt.sep)
                 {
-                    enum header = Unqual!(typeof(this)).stringof ~ "(",
-                         footer = ")",
-                         separator = ", ";
-                    sink(header);
                     foreach (i, Type; Types)
                     {
                         static if (i > 0)
                         {
-                            sink(separator);
+                            writer(fmt.sep);
                         }
-                        // TODO: Change this once formatElement() works for shared objects.
+                        // TODO: Change this once formattedWrite() works for shared objects.
                         static if (is(Type == class) && is(Type == shared))
                         {
-                            sink(Type.stringof);
+                            writer(Type.stringof);
                         }
                         else
                         {
-                            FormatSpec!Char f;
-                            formatElement(sink, field[i], f);
+                            formattedWrite(writer, fmt.nested, this.field[i]);
                         }
                     }
-                    sink(footer);
                 }
                 else
                 {
-                    throw new FormatException(
-                        "Expected '%s' or '%(...%)' or '%(...%|...%)' format specifier for type '" ~
-                            Unqual!(typeof(this)).stringof ~ "', not '%" ~ fmt.spec ~ "'.");
+                    formattedWrite(writer, fmt.nested, staticMap!(sharedToString, this.expand));
+                }
+            }
+            else if (fmt.spec == 's')
+            {
+                enum header = Unqual!(typeof(this)).stringof ~ "(",
+                     footer = ")",
+                     separator = ", ";
+                writer(header);
+                foreach (i, Type; Types)
+                {
+                    static if (i > 0)
+                    {
+                        writer(separator);
+                    }
+                    // TODO: Change this once formatElement() works for shared objects.
+                    static if (is(Type == class) && is(Type == shared))
+                    {
+                        writer(Type.stringof);
+                    }
+                    else
+                    {
+                        FormatSpec!Char f;
+                        formatElement(writer, field[i], f);
+                    }
                 }
+                writer(footer);
+            }
+            else
+            {
+                throw new FormatException(
+                    "Expected '%s' or '%(...%)' or '%(...%|...%)' format specifier for type '" ~
+                        Unqual!(typeof(this)).stringof ~ "', not '%" ~ fmt.spec ~ "'.");
```

Comment:
AFAICT this isn't related to this PR. Please move it into a separate PR. Thanks!

Reply:
I'll do.